flipper-plugin-realm/src/index.tsx

```diff
@@ -81,82 +83,79 @@ export function plugin(client: PluginClient<Events, Methods>) {
 
   client.onMessage('liveObjectAdded', (data: AddLiveObjectRequest) => {
     const state = pluginState.get();
-    const { index, schema, newObject, newObjectKey } = data;
-    if (schema != state.currentSchema?.name) {
+    const { index, schema, newObject } = data;
+    if (schema !== state.currentSchema?.name) {
       return;
     }
-    if (index > state.objects.length) {
-       pluginState.set({
-         ...state,
-         totalObjects: state.totalObjects + 1,
-       });
+ 
+    // Object already inserted
+    if (index < state.objects.length && state.objects[index]._pluginObjectKey == newObject._pluginObjectKey) {
       return;
     }
     const clone = structuredClone(newObject);
-    clone._objectKey = newObjectKey;
     const copyOfObjects = state.objects;
     const addedObject = convertObjects(
       [clone],
       state.currentSchema,
-      downloadData
+      downloadData,
     )[0];
     copyOfObjects.splice(index, 0, addedObject);
     const newLastObject = copyOfObjects[copyOfObjects.length - 1];
     pluginState.set({
       ...state,
       objects: [...copyOfObjects],
       totalObjects: state.totalObjects + 1,
-      cursor: newLastObject._objectKey as string,
+      cursor: newLastObject._pluginObjectKey,
     });
   });
 
   client.onMessage('liveObjectDeleted', (data: DeleteLiveObjectRequest) => {
     const state = pluginState.get();
     const { index, schema } = data;
```

Comment:
If `schema` is just the name of the `schema`, then this should be renamed to `schemaName`.

Reply:
good point, this is a reoccuring naming issue throughout the codebase... I will fix it in this response/request but I think won't go deeper to avoid too much time refactoring for now.